### PR TITLE
Workspace: remove unnecessary flags on Windows

### PR DESCRIPTION
**Pull request by compnerd**
This cleans up the unnecessary flags now that they are not absolutely
required.  This will break compatibility with older releases, however,
because there is no current stable release on Windows with
swift-package-manager support, this is acceptable.

### CODE REVIEW

**Comment by compnerd** [comment]
I think that I still need a day or two for all the changes to propagate through everything, but, I think that this should correct the handling of flags.

**Review by abertelrud — APPROVED** [review]

**Comment by abertelrud** [comment]
@swift-ci please smoke test

**Comment by abertelrud** [comment]
@compnerd Do you think this is ready to merge (assuming the tests pass), or do you still want to hold off for a bit?

**Comment by compnerd** [comment]
I think that this should be ready to merge, once the CI is happy again.

**Comment by abertelrud** [comment]
@swift-ci please smoke test

**Comment by abertelrud** [comment]
The Linux test failed with: `error: pathspec 'master' did not match any file(s) known to git`.  This should be fixed by now.  Restarting test.

**Comment by abertelrud** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
Awesome!  Happy to get rid of the unnecessary flags that I had to pollute s-p-m with for a bit.  Thanks for the review and triggering the CI @abertelrud!

**Pull request merged**
